actionview/lib/action_view/test_case.rb

```diff
@@ -42,6 +42,176 @@ def initialize
+        # method, where +[FORMAT]+ corresponds to the value of the
+        # +format+ argument
+        #
+        # === Arguments
+        #
+        # <tt>format</tt> - Symbol the name of the format used to render view's content
+        # <tt>callable</tt> - Callable to parse the String. Accepts the String
+        #                     value as its only argument
+        # <tt>block</tt> - Block serves as the parser when the
+        #                  <tt>callable</tt> is omitted
+        #
+        # By default, ActionView::TestCase defines a parser for:
+        #
+        #   * :html - returns an instance of Nokogiri::XML::Node
+        #   * :json - returns an instance of ActiveSupport::HashWithIndifferentAccess
+        #
+        # Each pre-registered parser also defines a corresponding helper:
+        #
+        #   * :html - defines `rendered.html`
+        #   * :json - defines `rendered.json`
+        #
+        # === Examples
+        #
+        #   test "renders HTML" do
+        #     article = Article.create!(title: "Hello, world")
+        #
+        #     render partial: "articles/article", locals: { article: article }
+        #
+        #     assert_pattern { rendered.html.at("main h1") => { content: "Hello, world" } }
+        #   end
+        #
+        #   test "renders JSON" do
+        #     article = Article.create!(title: "Hello, world")
+        #
+        #     render formats: :json, partial: "articles/article", locals: { article: article }
+        #
+        #     assert_pattern { rendered.json => { title: "Hello, world" } }
+        #   end
+        #
+        # To parse the rendered content into RSS, register a call to <tt>RSS::Parser.parse</tt>:
+        #
+        #   register_parser :rss, -> rendered { RSS::Parser.parse(rendered) }
+        #
+        #   test "renders RSS" do
+        #     article = Article.create!(title: "Hello, world")
+        #
+        #     render formats: :rss, partial: article
+        #
+        #     assert_equal "Hello, world", rendered.rss.items.last.title
+        #   end
+        #
+        # To parse the rendered content into a Capybara::Simple::Node,
+        # re-register an <tt>:html</tt> parser with a call to
+        # <tt>Capybara.string</tt>:
+        #
+        #   register_parser :html, -> rendered { Capybara.string(rendered) }
+        #
+        #   test "renders HTML" do
+        #     article = Article.create!(title: "Hello, world")
+        #
+        #     render partial: article
+        #
+        #     rendered.html.assert_css "h1", text: "Hello, world"
+        #   end
+        def register_parser(format, callable = nil, &block)
+          parsers[format] = callable || block || :itself.to_proc
+        end
+      end
+
+      included do
+        class_attribute :parsers, instance_accessor: false, default: ActiveSupport::HashWithIndifferentAccess.new
+
+        ##
+        # :method: rendered.html
```

Comment:
I think we should document the rendered method and then document the methods available in that object